### PR TITLE
First pass `tsc --checkJs true` run

### PR DESCRIPTION
**Pull request by jelly**
Plus a small bug fix:

image

To

image

And now we close the Reboot dropdown after clicking on a dropdown item.

### CODE REVIEW

File: pkg/systemd/overview-cards/cryptoPolicies.jsx [hunk 1]
```diff
@@ -223,7 +223,7 @@ export const CryptoPolicyStatus = () => {
     if (isInconsistentPolicy(currentCryptoPolicy, fipsEnabled)) {
         return (
             <li className="system-health-crypto-policies">
-                <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }}>
```
Comment (martinpitt) [inline pkg/systemd/overview-cards/cryptoPolicies.jsx:226]: This also appears in two more places:

pkg/systemd/overview-cards/lastLogin.jsx:            <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }}>
pkg/systemd/overview-cards/shutdownStatus.jsx:            <Flex spacer={{ default: 'spaceItemsSm' }} flexWrap={{ default: 'nowrap' }}>

Plus, are these valid?

pkg/users/account-create-dialog.js:                    <FlexItem spacer={{ default: 'spacerNone' }}>
pkg/users/account-create-dialog.js:                    <FlexItem spacer={{ default: 'spacerNone' }}>
pkg/users/account-details.js:                                        <FlexItem spacer={{ default: 'spacerNone' }}>

**Review by martinpitt — APPROVED** [review]
Thanks!

If this goes green, ok to land. If you need to change anyway, please squash the commits a bit.

**Review by allisonkarlitskaya — COMMENTED** [review]
My main feedback here: these are some nice drive-bys, but it would be nice to know what you're driving towards.  I mean that in the sense of "what can we add to test/static-code, or our tsconfig, so that this category of errors can't be introduced again in the future".

I ran `tsc --checkJs true` as implied by the title of this PR and got 11009 errors.  That seems like something we'll never fix.  `--checkJs true --strict false` would catch the sort of errors you're fixing here, and is a more manageable 1992, with very many of those in the qunit tests.  That feels like something we might aim for...

File: pkg/packagekit/updates.jsx [hunk 2]
```diff
@@ -340,7 +340,7 @@ function updateItem(remarkable, info, pkgNames, key) {
     if (pkgNames.some(pkg => isKpatchPackage(pkg.name)))
         pkgsTruncated.push(
             <LabelGroup key={`${key}-kpatches-labelgroup`} className="kpatches-labelgroup">
-                {" "}<Badge color="blue" variant="filled">{_("patches")}</Badge>
+                {" "}<Badge color="blue">{_("patches")}</Badge>
```
Comment (cockpituous) [inline pkg/packagekit/updates.jsx:343]: This added line is not executed by any test.

**Pull request merged**